src/backend/distributed/executor/multi_utility.c

```diff
@@ -299,20 +409,41 @@ ProcessCopyStmt(CopyStmt *copyStatement, char *completionTag)
 	 */
 	if (copyStatement->relation != NULL)
 	{
-		Relation copiedRelation = NULL;
-		bool isDistributedRelation = false;
+		bool isDistributedTable = false;
+		bool workerCopy = WorkerCopy(copyStatement);
+
+		if (workerCopy)
+		{
+			NodeAddress *masterNodeAddress = MasterNodeAddress(copyStatement);
+			char *relationName = copyStatement->relation->relname;
+			char *nodeName = masterNodeAddress->nodeName;
+			int32 nodePort = masterNodeAddress->nodePort;
+
+			isDistributedTable = IsRemoteDistributedTable(relationName, nodeName,
+														  nodePort);
+			if (!isDistributedTable)
+			{
+				ereport(ERROR, (errmsg("relation \"%s\" on the master node is not a "
+									   "distributed table", relationName)));
+			}
 
-		copiedRelation = heap_openrv(copyStatement->relation, AccessShareLock);
+			CreateLocalTable(relationName, nodeName, nodePort);
```

Comment:
>
> What happens if the table exists?
>
>
This is not a problem, because we create new tables as temporary and
temporary tables exist in a special schema. Though, I need to think about
schema support a bit more and test it well.